core/src/main/scala/chisel3/package.scala

```diff
@@ -37,13 +39,48 @@ package object chisel3 {
       case bigint                => Builder.error(s"Cannot convert $bigint to Bool, must be 0 or 1"); Bool.Lit(false)
     }
 
-    /** Int to UInt conversion, recommended style for constants.
-      */
-    def U: UInt = UInt.Lit(bigint, Width())
+    /** Int to UInt conversion, recommended style for constants. */
+    def U: UInt = UInt.Lit(bigint, Width()) // scalastyle:ignore method.name
 
-    /** Int to SInt conversion, recommended style for constants.
-      */
-    def S: SInt = SInt.Lit(bigint, Width())
+    /** BigInt to UInt conversion followed by bit extract (bad style, deprecated) */
+    @chiselRuntimeDeprecated
+    @deprecated(
+      "Passing an Int to .U is usually a mistake: It does *not* set the width but does a bit extract. " +
+        "Did you mean .U(<arg>.W)? " +
+        "If you do want bit extraction, use .U.extract(<arg>)",
+      "3.5.3"
+    )
+    def U(x: BigInt): Bool = { // scalastyle:ignore method.name
+      // Note that both source locator and compile options are ignored for apply on literal
+      this.U.extract(x)(DeprecatedSourceInfo, ExplicitCompileOptions.Strict)
+    }
+
+    /** BigInt to UInt conversion followed by bit extract */
+    def U(x: BigInt, y: BigInt): UInt = { // scalastyle:ignore method.name
+      // Note that both source locator and compile options are ignored for apply on literal
+      this.U.do_apply(x, y)(DeprecatedSourceInfo, ExplicitCompileOptions.Strict)
+    }
+
```

Comment:
Should this version also be deprecated?

Reply:
It sounds like we agreed this version does not need to be deprecated? As we seem to think it is not ambiguous?

Reply:
👍